dpnp/fft/dpnp_utils_fft.py

```diff
@@ -395,13 +395,31 @@ def _fft(a, norm, out, forward, in_place, c2c, axes, batch_fft=True):
         a = dpnp.reshape(a, local_shape)
         index = 1
 
+        # cuFFT requires input arrays to be C-contiguous (row-major)
+        # for correct execution
+        if dpnp.is_cuda_backend(a) and not a.flags.c_contiguous:
+            a = dpnp.ascontiguousarray(a)
```

Comment:
The above call of `reshape` has `order` keyword which defaults to `"C"`.
Why is it required to have explicit separate call of `ascontiguousarray`? And why it is not granted by `reshape` call? Is it a bug or expected `reshape` behavior?

Reply:
In some cases we can have `a.shape=local shape` then `dpnp.reshape()` will return the array view (without copying) regardless of `order` parameter
```
a = dpnp.ones((1,2,3),order="F")
dpnp.reshape(a,(-1,2,3),order="C").flags
Out[46]: 
  C_CONTIGUOUS : False
  F_CONTIGUOUS : True
  WRITABLE : True

```

This is the expected behavior. Similar to NumPy